protocol_connect.go

```diff
@@ -49,6 +49,11 @@ const (
 	connectStreamingContentTypePrefix = "application/connect+"
 )
 
+// defaultConnectUserAgent returns a User-Agent string similar to those used in gRPC.
+//
+//nolint:gochecknoglobals
+var defaultConnectUserAgent = []string{fmt.Sprintf("connect-go/%s (%s)", Version, runtime.Version())}
```

Comment:
The only potential negative I could think about this is someone could potentially modify this slice in an interceptor (and impact all future calls). Very unlikely but that might be a reason to make this a const string and just alloc the slice each time.

Reply:
It can't be a const string since we need to build it at runtime. But yeah, someone would have to awkwardly do like, `Header["User-Agent"][0] = "..."` and it'd hijack it. Any other mutation such as append wouldn't since it'd create a new slice reference.

I'm fine with keeping it as `[]string{defaultConnectUserAgent}` and force a new allocation since the expensive part was the string formatting.

Reply:
I much prefer leaving nothing to chance with globals, so making a new string slice each time seems sensible to me...the global string slice feels like a future security issue somehow